docs/docs/components-and-props.md

```diff
@@ -0,0 +1,248 @@
+---
+id: components-and-props
+title: Components and Props
+permalink: docs/components-and-props.html
+prev: rendering-elements.html
+next: state-and-lifecycle.html
+---
+
+Components let you split the UI into independent, reusable pieces, and think about each piece in isolation.
+
+Conceptually, components are like JavaScript functions. They accept arbitrary inputs (called "props") and return React elements describing what should appear on the screen.
+
+## Functional and Class Components
+
+The simplest way to define a component is to write a JavaScript function:
+
+```js
+function Welcome(props) {
+  return <h1>Hello, {props.name}</h1>;
+}
+```
+
+This function is a valid React component because it accepts a single "props" object argument with data and returns a React element. We call such components "functional" because they are literally JavaScript functions.
+
+You can also use an [ES2015 class](https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Classes) to define a component:
+
+```js
+class Welcome extends React.Component {
+  render() {
+    return <h1>Hello, {this.props.name}</h1>;
+  }
+}
+```
+
+The above two components are equivalent from React's point of view.
+
+Classes have some additional features that we will discuss in the [next sections](/react/docs/state-and-lifecycle.html). Until then, we will use functional components for their conciseness.
+
+## Rendering a Component
+
+Previously, we only encountered React elements that represent DOM tags:
+
+```js
+const element = <div />;
+```
+
+However, elements can also represent user-defined components:
+
+```js
+const element = <Welcome name="Sara" />;
+```
+
+When React sees an element representing a user-defined component, it passes JSX attributes to this component as a single object. We call this object "props".
+
+For example, this code renders "Hello, Sara" on the page:
+
+```js{1,5}
+function Welcome(props) {
+  return <h1>Hello, {props.name}</h1>;
+}
+
+const element = <Welcome name="Sara" />;
+const container = document.getElementById('root');
+ReactDOM.render(element, container);
+```
+
+[Try it on Codepen.](http://codepen.io/gaearon/pen/YGYmEG?editors=0010)
+
+Let's recap what happens in this example:
+
+1. We call `ReactDOM.render(<Welcome name="Sara" />, container)`.
+3. React calls the `Welcome` component with `{name: 'Sara'}` as the props.
+4. Our `Welcome` component returns `<h1>Hello, Sara</h1>` element as the result.
+5. React DOM efficiently updates the DOM to match `<h1>Hello, Sara</h1>`.
+
+>**Caveat:**
+>
+>Always start component names with a capital letter. For example, `<div />` element represents a DOM tag, but `<Welcome />` represents a component. Also make sure your components are in scope when you are using them.
```

Comment:
remove "element" after `<div />` - i think. it is hard to make these mixed code-and-words sentences grammatical